energy-integration/farm-boosted-yields/src/lib.rs

```diff
@@ -44,31 +54,41 @@ pub trait FarmBoostedYieldsModule:
     + weekly_rewards_splitting::update_claim_progress_energy::UpdateClaimProgressEnergyModule
     + energy_query::EnergyQueryModule
 {
+    #[only_owner]
     #[endpoint(collectUndistributedBoostedRewards)]
-    fn collect_undistributed_boosted_rewards(&self) {
-        self.require_caller_has_admin_permissions();
-
-        let collect_rewards_offset = USER_MAX_CLAIM_WEEKS + 1usize;
+    fn collect_undistributed_boosted_rewards(
+        &self,
+        start_week: Week,
+        end_week: Week,
+        dest_address: ManagedAddress,
+    ) {
+        let collect_rewards_offset = USER_MAX_CLAIM_WEEKS + 1;
```

Comment:
Why use an offset? Why not claiming all the rewards from `start_week` to `end_week`?

Reply:
It was like this in the initial implementation and I kept it. It makes sense: Users can claim boosted rewards for the past 4 weeks anytime, so you should only collect rewards from 5 weeks onward.